addons/website_event_sale/models/sale_order.py

```diff
@@ -1,40 +1,39 @@
 # -*- coding: utf-8 -*-
-from openerp import SUPERUSER_ID
-from openerp.osv import osv
-from openerp.tools.translate import _
+from openerp import models, api, _
 from openerp.exceptions import UserError
 
 
-class sale_order(osv.Model):
+class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    def _cart_find_product_line(self, cr, uid, ids, product_id=None, line_id=None, context=None, **kwargs):
-        line_ids = super(sale_order, self)._cart_find_product_line(cr, uid, ids, product_id, line_id, context=context)
+    @api.multi
+    def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
+        line_ids = super(SaleOrder, self)._cart_find_product_line(product_id, line_id)
         if line_id:
             return line_ids
-        for so in self.browse(cr, uid, ids, context=context):
-            domain = [('id', 'in', line_ids)]
-            if context.get("event_ticket_id"):
-                domain += [('event_ticket_id', '=', context.get("event_ticket_id"))]
-            return self.pool.get('sale.order.line').search(cr, SUPERUSER_ID, domain, context=context)
-
-    def _website_product_id_change(self, cr, uid, ids, order_id, product_id, qty=0, line_id=None, context=None):
-        values = super(sale_order, self)._website_product_id_change(cr, uid, ids, order_id, product_id, qty=qty, line_id=line_id, context=None)
+        domain = [('id', 'in', line_ids)]
+        for so in self:
+            if self._context.get("event_ticket_id"):
```

Comment:
I would suggest `self.env.context`, it's slightly longer but probably more sensible.